_posts/2023-11-13-android-platforms.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: posts
+title: Android Platforms in Bazel 7.0
+authors:
+  - katre
+---
+
+Many rules in the Bazel ecosystem have migrated to using the
+[platforms](https://bazel.build/extending/platforms) and
+[toolchains](https://bazel.build/extending/toolchains) systems. We're excited to
+announce that, as of Bazel 7.0, the [built-in Android
+rules](https://bazel.build/reference/be/android) are joining this group.
+
+## TL;DR: What do I change?
+
+The short version is that instead of using `--fat_apk_cpu`, you'll use `--android_platforms` to indicate what specific device you want your Android APKs to build for.
+
+Just like the `--platforms` flag, the values passed to `--android_platforms` are
+the labels of [`platform`](https://bazel.build/reference/be/platforms-and-toolchains#platform) targets, using standard constraint values to describe your device.
+
+For example, for an Android device with a 64-bit ARM processor, you'd define
+your platform like this:
+
+```py
+platform(
+    name = "arm64",
+    constraint_values = [
+        "@platforms//os:android",
+        "@platforms//cpu:arm64",
+    ],
+)
+```
+
+Every Android `platform` should use the [`@platforms//os:android`](https://github.com/bazelbuild/platforms/blob/33a3b209f94856193266871b1545054afb90bb28/os/BUILD#L36) OS constraint. To migrate the CPU constraint, check this chart:
+
+CPU Value     | Platform
+------------- | ------------------------------------------
+`armeabi-v7a` | `@platforms//cpu:arm`
+`arm64-v8a`   | `@platforms//cpu:arm64`
+`x86`         | `@platforms//cpu:x86_32`
+`x86_64`      | `@platforms//cpu:x86_64`
+
+And, of course, for a multi-architecture APK, you pass multiple labels, for
+example: `--android_platforms=//:arm64,//:x86_64` (assuming you defined those in
+your top-level `BUILD.bazel` file).
+
+### Are any other flags changing?
+
+Yes, as part of this several legacy flags have been deprecated and are no-ops.
+
+Deprecated flags (and their new equivalents):
+
+- `--android_cpu`: Like `--fat_apk_cpu`, this is replaced by
+  `--android_platforms`
+- `--android_sdk`: Android SDKs are now registered toolchains, so if you have
+  multiple `android_sdk_repository` repository rules defined, you can use
+  `register_toolchains` and `--extra_toolchains` to change the order they are
+  used in.
+- `--android_crosstool_top`: Similarly to the Android SDK, the
+  `android_ndk_repository` rule exposes the NDK as a C++ toolchain, using the
+  standard toolchain resolution. You can use `register_toolchains` and
+  `--extra_toolchains` to change the selection order for these.
+
+### Can I test this out?
+
+Yes, download the latest [Bazel 7.0 release
+candidate](https://releases.bazel.build/7.0.0/rc3/index.html) and update your
```

Comment:
Nit: We plan to cut a new release candidate likely tomorrow.

Perhaps we should mention that with bazelisk / `last_rc` you get the most recent RC?

Reply:
Too bad we don't have a central page for release candidates. Added a link for Bazelisk.

Reply:
cc @fweikert as we talked about this earlier